### PR TITLE
CLB: Background mechanic and related cards

### LINKED ISSUE #251: CLB: Choose Background + Background Type
From what is leaked, Background works as similar to Partner in a way like Oath Breaker.

You can have a Legendary with "Choose a Background" as Commander
and a Legendary Enchantment - Background as second Commander

from what it looks like, the choice isn't flexible, but done like Partner Commander while Deck Building

GUI wise the CanBeCommander checks needs to be extended to allow for secondary choices.
Meaning you need to first choose a Commander with "Choose a Background", then you need to choose the Background card

### LINKED ISSUE #526: Cloakwood Hermit

### LINKED ISSUE #527: Halsin, Emerald Archdruid

### LINKED ISSUE #529: Alora, Merry Thief

### LINKED ISSUE #530: Raised by Giants

### LINKED ISSUE #531: Master Chef

### LINKED ISSUE #534: Shameless Charlatan

### LINKED ISSUE #535: Livaan, Cultist of Tiamat

### PR DESCRIPTION
**Pull request by Northmoc**
- closes #535 
- closes #529 
- closes #530 
- closes #534 
- closes #531 
- closes #527 
- closes #526 
- closes #251

### CODE REVIEW

**Review by Hanmac — APPROVED** [review]
👍 

for the gui part, might be done later

File: forge-core/src/main/java/forge/card/CardRules.java [hunk 1]
```diff
@@ -215,7 +215,8 @@ public String getOracleText() {
     }
 
     public boolean canBeCommander() {
-        if (mainPart.getOracleText().contains("can be your commander")) {
+        if (mainPart.getOracleText().contains("can be your commander")
+                || mainPart.getType().hasSubtype("Background")) {
```
Comment (tool4ever) [inline forge-core/src/main/java/forge/card/CardRules.java:219]: you can reuse `canBeBackground` here

File: forge-gui/res/cardsfolder/upcoming/livaan_cultist_of_tiamat.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Name:Livaan, Cultist of Tiamat
+ManaCost:2 R
+Types:Legendary Creature Dragon Shaman
+PT:1/3
+T:Mode$ SpellCast | ValidCard$ Card.nonCreature | ValidActivatingPlayer$ You | Execute$ TrigPump | TriggerZones$ Battlefield | TriggerDescription$ Whenever you cast a noncreature spell, target creature gets +X/+0 until end of turn, where X is that spell's mana value.
+SVar:TrigPump:DB$ Pump | ValidTgts$ Creature.YouCtrl | TgtPrompt$ Select target creature you control | NumAtt$ +X
+SVar:X:TriggeredCard$CardManaCost
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/livaan_cultist_of_tiamat.txt:7]: This should probably be `TriggerCount$CastSACMC`
(it matters if it's X spell that gets countered before trigger resolves)

File: forge-gui/res/cardsfolder/upcoming/master_chef.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Name:Master Chef
+ManaCost:2 G
+Types:Legendary Enchantment Background
+S:Mode$ Continuous | Affected$ Creature.IsCommander+YouOwn | AddReplacementEffects$ This & Other | Description$ Commander creatures you own have "This creature enters the battlefield with an additional +1/+1 counter on it" and "Other creatures you control enter with an additional +1/+1 counter on them."
+SVar:This:Event$ Moved | ValidCard$ Card.Self | Destination$ Battlefield | ReplaceWith$ ExtraCounter | Description$ This creature enters the battlefield with an additional +1/+1 counter on it.
+SVar:Other:Event$ Moved | ValidCard$ Creature.Other+YouCtrl | Destination$ Battlefield | ReplaceWith$ ExtraCounter | Description$ Other creatures you control enter with an additional +1/+1 counter on them.
+SVar:ExtraCounter:DB$ PutCounter | ETB$ True | Defined$ ReplacedCard | CounterType$ P1P1 | SubAbility$ ETB
+SVar:ETB:DB$ ChangeZone | Origin$ All | Destination$ Battlefield | Defined$ ReplacedCard
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/master_chef.txt:8]: @Hanmac 
shouldn't this use InternalEtbReplacement?

Comment (Hanmac) [reply forge-gui/res/cardsfolder/upcoming/master_chef.txt:8]: yeah use InternalEtbReplacement for now

you also can still use etbCounter until i remove it

**Review by Hanmac — APPROVED** [review]
👍

**Pull request merged**